kimchi/src/circuits/constraints.rs

```diff
@@ -649,6 +657,45 @@ impl<F: FftField + SquareRootField> ConstraintSystem<F> {
         // TODO: This doesn't need to be degree 8 but that would require some changes in expr
         let coefficients8 = array_init(|i| coefficientsm[i].evaluate_over_domain_by_ref(domain.d8));
 
+        // Cairo constraint selector polynomials
+        let cairo: [DP<F>; turshi::CIRCUIT_GATE_COUNT] = array_init(|i| {
+            let g = match i {
+                0 => GateType::CairoClaim,
+                1 => GateType::CairoInstruction,
+                2 => GateType::CairoFlags,
+                3 => GateType::CairoTransition,
+                _ => panic!("Invalid index"),
+            };
+            E::<F, D<F>>::from_vec_and_domain(
+                gates
+                    .iter()
+                    .map(|gate| if gate.typ == g { F::one() } else { F::zero() })
+                    .collect(),
+                domain.d1,
+            )
+            .interpolate()
+        });
+
+        // Cairo constraint polynomials
+        let cairo8: [E<F, D<F>>; turshi::CIRCUIT_GATE_COUNT] = array_init(|i| {
+            let g = match i {
+                0 => GateType::CairoClaim,
+                1 => GateType::CairoInstruction,
+                2 => GateType::CairoFlags,
+                3 => GateType::CairoTransition,
+                _ => panic!("Invalid index"),
+            };
+            E::<F, D<F>>::from_vec_and_domain(
+                gates
+                    .iter()
+                    .map(|gate| if gate.typ == g { F::one() } else { F::zero() })
+                    .collect(),
+                domain.d1,
+            )
+            .interpolate()
+            .evaluate_over_domain_by_ref(domain.d8)
+        });
```

Comment:
move the cairo initialization to its own function which you can call here :o

Reply:
I am not sure I get it. None of the other gates have an independent function for this, do they?

Reply:
if you look at lookup it does

Reply:
see https://github.com/o1-labs/proof-systems/blob/master/kimchi/src/circuits/constraints.rs#L345

Reply:
Oh I see, for lookup yeah... I was looking at chacha for instance. But fine, I can make it separately.